webapi/src/main/scala/org/knora/webapi/util/ConstructResponseUtilV2.scala

```diff
@@ -521,6 +416,183 @@ object ConstructResponseUtilV2 {
+            resourceIri =>
+                val transformedResource = nestResources(
+                    resourceIri = resourceIri,
+                    flatResourcesWithValues = flatResourcesWithValues,
+                    visibleResources = visibleResources,
+                    dependentResourceIrisVisible = dependentResourceIrisVisible,
+                    dependentResourceIrisNotVisible = dependentResourceIrisNotVisible,
+                    incomingLinksForResource = incomingLinksForResource
+                )
+
+                resourceIri -> transformedResource
+        }.toMap
+
+        MainResourcesAndValueRdfData(
+            resources = mainResourcesNested,
+            hiddenResourceIris = mainResourceIrisNotVisible ++ dependentResourceIrisNotVisible
+        )
+    }
+
+    /**
+     * This method creates a map of property IRIs to sequences of value IRIs.
+     *
+     * @param assertionsExplicit
+     * @param valueObjectIris           set of all value object IRIs
+     * @return                          the sequence of IRIs
+     */
+    private def mapIriSequence(assertionsExplicit: ConstructPredicateObjects, valueObjectIris: Set[IRI] ) :  Map[SmartIri, Seq[IRI]] =  {
+        val valuePropertyToObjectIris: Map[SmartIri, Seq[IRI]] = assertionsExplicit.map {
+            case (pred: SmartIri, objs: Seq[LiteralV2]) =>
+                // Get only the assertions in which the object is a value object IRI.
+
+                val valueObjIris: Seq[IriLiteralV2] = objs.collect {
+                    case iriObj: IriLiteralV2 if valueObjectIris(iriObj.value) => iriObj
+                }
+
+                pred -> valueObjIris
+        }.filter {
+            case (_: SmartIri, objs: Seq[IriLiteralV2]) => objs.nonEmpty
+        }.groupBy {
+            case (pred: SmartIri, _: Seq[IriLiteralV2]) =>
+                // Turn the sequence of assertions into a Map of predicate IRIs to assertions.
+                pred
+        }.map {
+            case (pred: SmartIri, valueAssertions: Map[SmartIri, Seq[IriLiteralV2]]) =>
+                // Replace the assertions with their objects, i.e. the value object IRIs.
+                pred -> valueAssertions.values.flatten.map(_.value).toSeq
+        }
+
+        return valuePropertyToObjectIris
+    }
+
+
+    /**
+     * This method creates a map of IRI sequences which is used to create a map of Rdf Data
+     *
+     * @param valuePropertyToObjectIris     the map of sequence IRIs
+     * @param resourceIri                   the IRI of the resource.
+     * @param requestingUser                the user making the request.
+     * @param nonResourceStatements         the statements which got split up containing value objects and standoffs
+     * @return                              the map of IRI sequences
+     */
+    private def mapSequenceObjects(valuePropertyToObjectIris: Map[SmartIri, Seq[IRI]], resourceIri:IRI, requestingUser: UserADM,
```

Comment:
Could this method's name be made clearer? For example: `makePropertyIriToValueObjectsMap`?